helix-core/src/history.rs

```diff
@@ -133,6 +133,32 @@ impl History {
         Some(&self.revisions[last_child.get()].transaction)
     }
 
+    // Get the position of last change
+    pub fn last_edit_pos(&self) -> Option<usize> {
+        let current_revision = &self.revisions[self.current];
+        let mut primary_selection = 0;
+        if let Some(tsel) = current_revision.transaction.selection() {
+            if let Some(isel) = current_revision.inversion.selection() {
```

Comment:
We only need the inversion here which will always have a selection (what the selection was in the document before the change was made).


```rust
pub fn last_edit_pos(&self) -> Option<usize> {
    let current_revision = &self.revisions[self.current];
    let primary_selection = current_revision.inversion.selection().expect("inversion always contains a selection").primary();
    let (from, _to, fragment) = current_revision.transaction
        .changes_iter()
        .find(|range| range.overlaps(primary_selection)) // find a change that matches the primary selection
        .or(current_revision.transaction.changes_iter()) // or use the first change
        .unwrap();
    let len = fragment.map(|text| text.chars.count()).unwrap_or(0);
    Some(from + len)
}
```